docs/agents/agent-development.md

```diff
@@ -237,14 +237,33 @@ Global labels can be specified via the environment variable `ELASTIC_APM_GLOBAL_
 
 ### Transactions
 
+#### Transaction outcome
+
+The `outcome` property denotes whether the transaction represents a success or a failure from the perspective of the entity that produced the event.
+The APM Server converts this to the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) field.
```

Comment:
The suggestion for different semantics for `transactions` vs. `spans`  for `event.outcome` makes sense and fits the description. What concerns me is that we do not have any APM/tracing related values for the other event categorization fields, and afaiu those fields have a hierarchical order. My question is whether we should start filling the lowest level value without specifying the other categorization values?

Reply:
It's a good point but I think adding `event.{kind|category|type}` is an orthogonal effort that isn't a prerequisite for `event.outcome`. Although certainly being important fields, I don't think we'd have an immediate use-case for them. But we do need a field to denote the success/error state of transactions and spans short term.
It's probably also something APM Server can determine without agents having to set an additional intake API field. Currently, it seems to be mostly focussed on security events.